### PR TITLE
Remove from cluster

### PR DESCRIPTION
**Pull request by ludwikbukowski**
This PR addresses https://github.com/esl/MongooseIM/issues/739

Proposed changes include:
* restored `remove_from_cluster` command. Use it in purpose to remove dead <strike>(and only dead)</strike> node from mongoose cluster.
* Added new node `ejabberd3@localhost` to travis architecture
* updated and added new tests in `cluster_commands_SUITE`
* updated docs

**[EDIT]**
After disscusion with _mim_ team I will modify the `remove_from_cluster` command so that it will be also possible to force remove _alive_ node.

### CODE REVIEW

**Comment by ludwikbukowski** [comment]
Only one test doesn't pass:
`compontent_SUITE:register_in_cluster/1`
after adding `ejabberd3@localhost` to travis architecture.
I'm trying to figure out why

File: test/ejabberd_tests/test.config [hunk 1]
```diff
@@ -31,6 +31,10 @@
                  {domain, <<"localhost">>},
                  {vars, "node2_vars.config"},
                  {cluster, mim}]},
+         {mim3, [{node, ejabberd3@localhost},
```
Comment (michalwski) [inline test/ejabberd_tests/test.config]: Do we really want to name the new node with `ejabberd` prefix?

File: test/ejabberd_tests/tests/cluster_commands_SUITE.erl [hunk 2]
```diff
@@ -308,14 +403,21 @@ normalize_args(Args) ->
...
-    run_interactive(Cmd, Response, timer:seconds(30)).
+    run_interactive(Cmd, Response, timer:seconds(60)).
 
 run_interactive(Cmd, Response, Timeout) ->
     Port = erlang:open_port({spawn, Cmd}, [exit_status]),
     %% respond to interactive question (yes/no)
     Port ! {self(), {command, Response}},
     loop(Port, [], Timeout).
 
+run(Cmd) ->
```
Comment (michalwski) [inline test/ejabberd_tests/tests/cluster_commands_SUITE.erl]: I see similar function in `ejabberd_ctl_helper`. Can we reuse it here?

Comment (ludwikbukowski) [inline test/ejabberd_tests/tests/cluster_commands_SUITE.erl]: yes we can, I will also reuse `normalize_args/1` and `loop/3` from the `ejabberdctl_helper` module

File: test/ejabberd_tests/test.config (hunk 1 above)
Comment (ludwikbukowski) [inline test/ejabberd_tests/test.config]: I'd love to name it with `mongooseim` prefix. I feel free to rename also `ejabberd2@localhost` node to `mongooseim2@localhost`. Is there any reason why not to do that?

Comment (michalwski) [inline test/ejabberd_tests/test.config]: Yes, some tests may fail because of that :) so that's work for separate PR I think.

**Pull request merged**